testsuite/tests/typing-misc/typecore_errors.ml

```diff
@@ -0,0 +1,477 @@
+Error: Unbound instance variable y
+|}]
+
+
+(** Not a packed type *)
+module type empty = sig  end
+let f (x:int) = ()
+let x = f (module struct end)
+[%%expect {|
+module type empty = sig  end
+val f : int -> unit = <fun>
+Line _, characters 10-29:
+  let x = f (module struct end)
+            ^^^^^^^^^^^^^^^^^^^
+Error: This expression is packed module, but the expected type is
+int
+|}]
+
+
+(** Bultin [%extension_constructor *)
+type t = A
+let x = [%extension_constructor A]
+[%%expect {|
+type t = A
+Line _, characters 32-33:
+  let x = [%extension_constructor A]
+                                  ^
+Error: This constructor is not an extension constructor.
+|}]
+
+let x = [%extension_constructor]
+[%%expect {|
+Line _, characters 8-32:
+  let x = [%extension_constructor]
+          ^^^^^^^^^^^^^^^^^^^^^^^^
+Error: Invalid [%extension_constructor] payload, a constructor is expected.
+|}]
+
+(** Invalid format *)
+let x = format_of_string "%z"
+[%%expect {|
+Line _, characters 25-29:
+  let x = format_of_string "%z"
+                           ^^^^
+Error: invalid format "%z": at character number 1, invalid conversion "%z"
+|}]
+
+(** Apply wrong label *)
+
+let f ~x = x + 2
+let y = f ~y:1
+[%%expect {|
+val f : x:int -> int = <fun>
+Line _, characters 13-14:
+  let y = f ~y:1
+               ^
+Error: The function applied to this argument has type x:int -> int
+This argument cannot be applied with label ~y
+|}]
+
+let g f = f ~x:0 ~y:0; f ~y:0 ~x:0
+[%%expect {|
+Line _, characters 23-24:
+  let g f = f ~x:0 ~y:0; f ~y:0 ~x:0
+                         ^
+Error: This function is applied to arguments
+in an order different from other calls.
+This is only allowed when the real type is known.
+|}]
+
+(** Inlined record *)
+type t = A of { x: int }
+let x = A 1
+[%%expect {|
+type t = A of { x : int; }
+Line _, characters 8-11:
+  let x = A 1
+          ^^^
+Error: This constructor expects an inlined record argument.
+|}]
+
+(** Illegal let rec *)
+type 'a t = A of 'a
+let rec A x = A (A ())
+
+[%%expect {|
+type 'a t = A of 'a
+Line _, characters 8-11:
+  let rec A x = A (A ())
+          ^^^
+Error: Only variables are allowed as left-hand side of `let rec'
+|}]
+
+(** Non-linear pattern *)
+
+let quadratic (x,x) = x * x
+[%%expect {|
+Line _, characters 17-18:
+  let quadratic (x,x) = x * x
+                   ^
+Error: Variable x is bound several times in this matching
+|}]
+
+(** Or-patter clash *)
+type t = A of int | B of float|C
+let f (A x|B x) = 0
+[%%expect {|
+type t = A of int | B of float | C
+Line _, characters 6-15:
+  let f (A x|B x) = 0
+        ^^^^^^^^^
+Error: The variable x on the left-hand side of this or-pattern has type
+       int but on the right-hand side it has type float
+|}]
+
+(** Orphan pattern variable *)
+
+let f (A x|C) = 0
+[%%expect {|
+Line _, characters 6-13:
+  let f (A x|C) = 0
+        ^^^^^^^
+Error: Variable x must occur on both sides of this | pattern
```

Comment:
Not your fault but the error message is weird, I would expect `or-pattern` instead of using a symbol in the middle of a sentence.